tests/testthat/test-as.network.numeric.R

```diff
@@ -0,0 +1,81 @@
+  expect_error(as.network.numeric(x=5, loops = TRUE, directed = TRUE))
+  expect_error(as.network.numeric(x=5, loops = TRUE, directed = FALSE))
+})
+test_that("Stop when the multiple == TRUE", {
+  expect_error(as.network.numeric(x=5, multiple = TRUE, bipartite = NULL))
+  expect_error(as.network.numeric(x=5, multiple = TRUE, directed = TRUE))
+  expect_error(as.network.numeric(x=5, multiple = TRUE, directed = FALSE))
+})
+test_that("Stop when the hyper == TRUE", {
+  expect_error(as.network.numeric(x=5, hyper = TRUE, bipartite = NULL))
+  expect_error(as.network.numeric(x=5, hyper = TRUE, directed = TRUE))
+  expect_error(as.network.numeric(x=5, hyper = TRUE, directed = FALSE))
+})
+test_that("Stop when the density is out of [0,1]", {
+  expect_error(as.network.numeric(x=5, density = -0.1, bipartite = NULL))
+  expect_error(as.network.numeric(x=5, density = -0.1, directed = TRUE))
+  expect_error(as.network.numeric(x=5, density = -0.1, directed = FALSE))
+  expect_error(as.network.numeric(x=5, density =  1.1, bipartite = NULL))
+  expect_error(as.network.numeric(x=5, density =  1.1, directed = TRUE))
+  expect_error(as.network.numeric(x=5, density =  1.1, directed = FALSE))
+})
+test_that("Stop when the number of edges is not integer", {
+  expect_error(as.network.numeric(x=5, numedges =  2.5, bipartite = NULL))
+  expect_error(as.network.numeric(x=5, numedges =  2.5, directed = TRUE))
+  expect_error(as.network.numeric(x=5, numedges =  2.5, directed = FALSE))
+})
+
+## # TODO: Add this back when explicit directed && bipartite is an error.
+##
+## test_that("Stop when a Directed bipartite network is specified by the user", {
+##   expect_error(as.network.numeric(x=5, numedges =  6L, directed = TRUE, bipartite = 3))
+## })
+
+#test_that("Stop when it is missed by the user to specify whether the network is bipartite or directed", {
+#  expect_error(as.network.numeric(x=5, numedges =  6L))
+#})
+
+## # TODO: Add this back when warnings can be relied on.
+## test_that("Warning message when bipartite == TRUE and user missed to specify whether the network is directed or undirected", {
+##   expect_warning(as.network.numeric(x=5, numedges =  6L, bipartite = 3))
+## })
+
+test_that("Correct number of edges for bipatite graph when numedges is passed", {
+  expect_silent(as.network.numeric(x=5, bipartite = 3, directed = FALSE))
+})
+test_that("Correct number of edges for undirected graph when numedges is passed", {
+  expect_silent(as.network.numeric(x=5, directed = FALSE))
+})
+test_that("Correct number of edges for directed graph when numedges is passed", {
+  expect_silent(as.network.numeric(x=5, directed = TRUE))
+})
+test_that("Correct number of edges for complete graph or when density == 1", {
+  expect_equal(network.edgecount(as.network.numeric(x=5, directed = FALSE, density = 1)),10)
+})
+test_that("Correct number of nodes for generated graph (numedges is less than 2^(53-4))", {
+  expect_error(as.network.numeric(x=2^50,directed = FALSE))
+})
```

Comment:
This test doesn't test what it says it tests. @AryaKarami, can you please fix this?

Reply:
Thanks for the comment. Clearly, the generated networks have the same number of nodes that the user feeds to function. So, I fixed the comment as follows. Is it correct?

test_that("Get the Correct number of nodes as the input, for which the numedges is less than 2^(53-4)", {
  expect_error(as.network.numeric(x=2^50,directed = FALSE))
})

Reply:
I think this test tests that specifying a bigger network that the function can generate results in an error?

Reply:
Yes, That is. I edited it as follows.

test_that("specifying a bigger network that the function can generate results in an error",{
expect_error(as.network.numeric(x=2^50,directed = FALSE))
})